SNIClient.py

```diff
@@ -1087,14 +1088,125 @@ async def main():
             time.sleep(3)
             sys.exit()
         elif args.diff_file.endswith((".apbp", "apz3")):
-            adjustedromfile, adjusted = Utils.get_adjuster_settings(romfile, gui_enabled)
-            if adjusted:
-                try:
-                    shutil.move(adjustedromfile, romfile)
-                    adjustedromfile = romfile
-                except Exception as e:
-                    logging.exception(e)
-            asyncio.create_task(run_game(adjustedromfile if adjusted else romfile))
+            lastSettings = Utils.get_adjuster_settings("LttP")
+            if lastSettings:
+                choice = 'no'
+                adjusted = False
+                if not hasattr(lastSettings, 'auto_apply') or 'ask' in lastSettings.auto_apply:
+
+                    whitelist = {"music", "menuspeed", "heartbeep", "heartcolor", "ow_palettes", "quickswap",
+                            "uw_palettes", "sprite", "sword_palettes", "shield_palettes", "hud_palettes",
+                            "reduceflashing", "deathlink"}
+                    printed_options = {name: value for name, value in vars(lastSettings).items() if name in whitelist}
+                    if hasattr(lastSettings, "sprite_pool"):
+                        sprite_pool = {}
+                        for sprite in getattr(lastSettings, "sprite_pool"):
+                            if sprite in sprite_pool:
+                                sprite_pool[sprite] += 1
+                            else:
+                                sprite_pool[sprite] = 1
+                        if sprite_pool:
+                            printed_options["sprite_pool"] = sprite_pool
+                    import pprint
+
+                    if gui_enabled:
+            
+                        from tkinter import Tk, PhotoImage, Label, LabelFrame, Frame, Button
```

Comment:
this might be best to extract out as a function, it makes following the flow of the outer function somewhat difficult